python/arcticdb/version_store/_normalization.py

```diff
@@ -1016,7 +1023,13 @@ def _nested_msgpack_unpackb(buff, raw=False):
 
     def _custom_pack(self, obj):
         if isinstance(obj, pd.Timestamp):
-            tz = obj.tz.zone if obj.tz is not None else None
+            if obj.tz and hasattr(obj.tz, "zone"):
+                tz = obj.tz.zone
+            else:
+                tz = obj.tzname()
+                # Edge case: Pandas tzname() returns "Coordinated Universal Time" instead of "UTC" on Windows
+                # Which is breaking the pytz deserialization used by msgpack
+                tz = "UTC" if tz == "Coordinated Universal Time" else tz
```

Comment:
Doesn't this mean that writes with tz == "Coordinated Universal Time" won't roundtrip correctly?

Reply:
According to the docs, it looks like Windows is passing `Coordinated Universal Time` into the `timezone` constructor.
https://docs.python.org/3/library/datetime.html#datetime.timezone.tzname

Indeed the name is arbitrary, and probably coming from a system local timezone database?

What's the use case for changing this?  Why not roundtrip, so the user gets back what they put in?

Reply:
If I understanding what's happening here, then it looks like we've hit a snag because it's not correctly round-tripping at the moment, because we didn't keep the offset.  The other possibility is that your testing is across OS environments, and the timezone is not usable in the other environment. 

If it's the first problem then I think there is a chance to fix it.   The de-serializer is reading just the first two elements of the tuple.  
https://github.com/man-group/ArcticDB/blame/5b6df4fdc7289fcd4fd0cb9601e4ddfc6f46b979/python/arcticdb/version_store/_normalization.py#L1056C9-L1056C9

We could serialize the offset as a third element, and then construct the `timezone` object ourselves.  De-serialisers from old versions will not be effected.

Reply:
This is a very specific edge case, which I think is a bug in pandas.
The scenario is (du == dateutil):
1. we create a `tz = tzwin('UTC')` with `du.tz.gettz("UTC")`
2. This is passed to pandas to create a pd.Timestamp - `ts = pd.Timestamp("2024-03-04", tz=tz)`
3. `ts.tzname()` returns `Coordinated Universal Time` instead of `UTC` which we have asked for
  - this happens **only on Windows and only with du.tz.gettz** , in any other case, trying to set the time zone to `UTC`, would results to it being `UTC` (actually there is no other way to create a tz with `Coordinated Universal Time` as it is not supported by pytz)
4. We save the result from `ts.tzname()` during the normalization/serialization
5. When we try to read the metadata, we get back `Coordinated Universal Time` and we try to create a pd.Timestamp with it
6. This fails because `Coordinated Universal Time` is `not a valid time zone name`

Actually,  `Coordinated Universal Time` is not a valid time zone name for `du.tz.gettz` as well, passing **that string to it also results in a crash**.
So the 2 ways for us to support this edge case would be to:
1. pickle the time zone data - which could cause other issues
2. store the `UTC` time zone during normalization, then during the denormalization, try to create the timezone with `du.tz.gettz("UTC")` and pass it to the pd.Timestamp, but only on Windows and we would need to know that it was `dateutil` to begin with

In short, I don't think that it is worth investing the time to support this edge case behavior that looks like a bug in pandas in the first place.

Reply:
You're explaining what's happening with `tzname` and this case.  My point is that we shouldn't be trying to interpret the `tzname`, we should be trying to get the original `pd.Timestamp` object back.  Which means we should be capturing `offset` and `name` for `timezone` and using e.g. ` pd.Timestamp('2024-03-04', tzinfo=timezone(timedelta(hours=0), name='Coordinate Universal time'))`, that is what the original `pd.Timestamp` contained.  What did the original `pd.Timestamp` contain?

Reply:
As discussed separately.  I understand now that we have a range of `tzinfo` subtypes in `pd.Timestamp` and that we're not going to support some of them, particularly `tzfile`.  So we do need to normalise the types coming out.